### PR TITLE
Adding a file upload api to the MSBLOC.Web project

### PR DESCRIPTION
**Pull request by profet23**
File upload API

### CODE REVIEW

File: MSBLOC.Web/Controllers/api/FileController.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            _logger = logger;
+            _tempFileService = tempFileService;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Upload()
+        {
+            if (!MultipartRequestHelper.IsMultipartContentType(Request.ContentType))
+            {
+                return BadRequest($"Expected a multipart request, but got {Request.ContentType}");
```
Comment (zippy1981) [inline MSBLOC.Web/Controllers/api/FileController.cs]: Not covered by a test.

File: MSBLOC.Web/Controllers/api/FileController.cs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                {
+                    if (MultipartRequestHelper.HasFileContentDisposition(contentDisposition))
+                    {
+                        var fileName = contentDisposition.FileName.Value;
+                        var path = await _tempFileService.CreateFromStreamAsync(fileName, section.Body);
+
+                        _logger.LogInformation($"Copied the uploaded file '{fileName}' to path: '{path}'");
+                    }
+                    else if (MultipartRequestHelper.HasFormDataContentDisposition(contentDisposition))
+                    {
```
Comment (zippy1981) [inline MSBLOC.Web/Controllers/api/FileController.cs]: Not covered by a test.

File: MSBLOC.Web/Controllers/api/FileController.cs [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                        }
+                    }
+                }
+
+                // Drains any remaining section body that has not been consumed and
+                // reads the headers for the next section.
+                section = await reader.ReadNextSectionAsync();
+            }
+
+            return new OkResult();
```
Comment (zippy1981) [inline MSBLOC.Web/Controllers/api/FileController.cs]: Should really be NoContentResult (204)

File: MSBLOC.Web/Controllers/api/FileController.cs [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            // most cases.
+            if (!hasMediaTypeHeader || Encoding.UTF7.Equals(mediaType.Encoding))
+            {
+                return Encoding.UTF8;
+            }
+            return mediaType.Encoding;
+        }
+    }
+
+    public class MultipartRequestHelper
```
Comment (zippy1981) [inline MSBLOC.Web/Controllers/api/FileController.cs]: Why is this not in its own file? Why no unit tests for it? Perhaps it should be internal.

File: MSBLOC.Web/Controllers/api/FileController.cs (hunk 1 above)
Comment (profet23) [reply MSBLOC.Web/Controllers/api/FileController.cs]: Test added.

File: MSBLOC.Web/Controllers/api/FileController.cs (hunk 2 above)
Comment (profet23) [reply MSBLOC.Web/Controllers/api/FileController.cs]: Test Added

**Comment by profet23** [comment]
> Can we remove unised namespaces? We all have resharper?

https://blogs.msdn.microsoft.com/csharpfaq/2004/10/20/be-mindful-of-the-references-using-manifest-relationship/


> Are we going to use temp file long term or use blogb storage etc eventually??

Depends on the lifetime of the files. The only reason they are being written to disk is because we don't want large files sit in memory. These files only exist for the life of the request. Then they are deleted. They should be moved to some other API before the request ends, but that API has yet to be written/determined.

**Pull request merged**